src/LibYear.Lib/FileTypes/XmlProjectFile.cs

```diff
@@ -1,37 +1,31 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using System.Xml.Linq;
-using NuGet.Versioning;
 
 namespace LibYear.Lib.FileTypes
 {
-    public abstract class XmlProjectFile : IProjectFile
+    public abstract class XmlProjectFile : XmlProject
     {
         private readonly string _elementName;
         private readonly string _packageAttributeName;
         private readonly string _versionAttributeName;
-        private readonly XDocument _xmlContents;
+        public override string FileName { get; }
 
-        public string FileName { get; }
-        public IDictionary<string, NuGetVersion> Packages { get; }
-
-        protected XmlProjectFile(string filename, string elementName, string packageAttributeName, string versionAttributeName)
+        protected XmlProjectFile(string filename, string elementName, string packageAttributeName, string versionAttributeName)

```

Comment:
It may be simpler to have this constructor take a stream (make it private if you want) and then use a factory method to construct from a file. `File.OpenRead` vs `new FileStream` for example.